source/beamline/BioXAS/BioXASCarbonFilterFarmActuatorPositionControl.cpp

```diff
@@ -71,38 +71,26 @@ void BioXASCarbonFilterFarmActuatorPositionControl::setMotor(CLSMAXvMotor *newCo
 {
 	if (motor_ != newControl) {
 
-		if (motor_)
-			removeChildControl(motor_);
+            removeChildControl(motor_);
+            motor_ = newControl;
+            addChildControl(motor_);
+            updateStates();
 
-		motor_ = newControl;
-
-		if (motor_)
-			addChildControl(motor_);
-
-		updateStates();
-
-		emit motorChanged(motor_);
+            emit motorChanged(motor_);
 	}
 }
 
 void BioXASCarbonFilterFarmActuatorPositionControl::setStatus(AMControl *newControl)
 {
 	if (status_ != newControl) {
 
-		if (status_)
-			removeChildControl(status_); // disconnects all signals.
-
-		status_ = newControl;
-
-		if (status_) {
-			addChildControl(status_);
-
-			connect( status_, SIGNAL(valueChanged(double)), this, SLOT(onStatusControlValueChanged()) );
-		}
-
-		updateStates();
+            removeChildControl(status_); // disconnects all signals.
+            status_ = newControl;
+            addChildControl(status_);
+            connect( status_, SIGNAL(valueChanged(double)), this, SLOT(onStatusControlValueChanged()) );
```

Comment:
@XuetaoLiu-CLS My concern here was changing behavior, since status would be status_ would be set to newControl regardless. If you think this is okay I'll go ahead and make the changes.